src/script/descriptor.cpp

```diff
@@ -1050,3 +1061,16 @@ std::unique_ptr<Descriptor> InferDescriptor(const CScript& script, const Signing
 {
     return InferScript(script, ParseScriptContext::TOP, provider);
 }
+
+void DescriptorCache::CacheExtPubKey(const KeyOriginInfo& origin_info, const CExtPubKey& xpub)
+{
+    m_xpubs[origin_info] = xpub;
```

Comment:
Does it make sense to `assert(m_xpubs.count(origin_info) == 0)`?

Reply:
Not at the moment: https://github.com/bitcoin/bitcoin/pull/18204#discussion_r384655169

Reply:
Performance penalty aside, makes sense to assert it's the same xpub if already cached?